test/performance/odp_l2fwd.c

```diff
@@ -487,17 +520,109 @@ static inline int process_extra_features(const appl_args_t *appl_args, odp_packe
 	return pkts;
 }
 
+static inline void handle_tx_event_compl(tx_compl_t *tx_c, odp_packet_t pkts[], int num,
+					 int tx_idx, stats_t *stats)
+{
+	odp_packet_t pkt;
+	int next = tx_c->n;
+	const int ival = tx_c->ival;
+
+	tx_c->opt.queue = gbl_args->pktios[tx_idx].compl_q;
+
+	while (next <= num) {
+		pkt = pkts[next - 1];
+
+		if (odp_packet_tx_compl_request(pkt, &tx_c->opt) < 0) {
+			stats->s.tx_c_fails++;
+			/* Missed one, try requesting for the first packet of next burst. */
+			next = num + 1;
+			break;
+		}
+
+		next += ival;
+	}
+
+	tx_c->n = next - num;
+}
+
+static inline void handle_tx_poll_compl(tx_compl_t *tx_c, odp_packet_t pkts[], int num, int tx_idx,
+					stats_t *stats)
+{
+	uint32_t compl = tx_c->compl, free = tx_c->free, next_free;
+	const uint32_t max = tx_c->max, init = tx_c->init;
+	odp_pktio_t pktio = gbl_args->pktios[tx_idx].pktio;
+	int next_pkt = tx_c->n;
+	odp_packet_t pkt;
+	const int ival = tx_c->ival;
+
+	while (compl != free) {
+		if (odp_packet_tx_compl_done(pktio, compl) != 1)
```

Comment:
The API specifies `>0` as transmit complete.

Reply:
V2.